pkg/chains/formats/slsa/v2alpha2/internal/resolved_dependencies/resolved_dependencies.go

```diff
@@ -0,0 +1,227 @@
+/*
+Copyright 2023 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resolveddependencies
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/in-toto/in-toto-golang/in_toto/slsa_provenance/common"
+	v1 "github.com/in-toto/in-toto-golang/in_toto/slsa_provenance/v1"
+	"github.com/tektoncd/chains/pkg/artifacts"
+	"github.com/tektoncd/chains/pkg/chains/formats/slsa/internal/material"
+	"github.com/tektoncd/chains/pkg/chains/objects"
+	"go.uber.org/zap"
+	"knative.dev/pkg/logging"
+)
+
+const (
+	// pipelineConfigName is the name of the resolved dependency of the pipelineRef.
+	pipelineConfigName = "pipeline"
+	// taskConfigName is the name of the resolved dependency of the top level taskRef.
+	taskConfigName = "task"
+	// pipelineTaskConfigName is the name of the resolved dependency of the pipeline task.
+	pipelineTaskConfigName = "pipelineTask"
+	// inputResultName is the name of the resolved dependency generated from Type hinted ARTIFACT_INPUTS results
```

Comment:
`ARTIFACT_INPUTS` is not the only type hinting.
```suggestion
	// inputResultName is the name of the resolved dependency generated from Type hinted parameters or results.
```